lib/embulk/input/marketo/lead.rb

```diff
@@ -42,35 +42,38 @@ def self.generate_columns(metadata)
         end
 
         def run
-          count = 0
           from_datetime = task[:from_datetime]
-          to_datetime = task[:to_datetime]
+          to_datetime = task[:to_datetime] || Time.now
+          ranges = task_target(from_datetime, to_datetime, task[:workers], index)
```

Comment:
time ranges should not be generated here. transaction method should create all time ranges and store them to task (task source). Here run method can get time ranges from task by using index.